app/src/androidTest/java/com/github/sdp/ratemyepfl/fragment/review/AddRestaurantReviewFragmentTest.kt

```diff
@@ -0,0 +1,144 @@
+        onView(withId(R.id.addReviewComment)).perform(typeText(comment))
+        closeSoftKeyboard()
+        onView(withId(R.id.doneButton)).perform(click())
+        onView(withId(R.id.addReviewComment)).check(matches(withText(comment)))
+    }
+
+    @Test
+    fun nullCommentNoReset() {
+        onView(withId(R.id.reviewNavigationView)).perform(CustomViewActions.navigateTo(R.id.addRestaurantReviewFragment))
+        val title = "Good"
+        onView(withId(R.id.reviewRatingBar)).perform(click())
+        onView(withId(R.id.addReviewTitle)).perform(typeText(title))
+        closeSoftKeyboard()
+        onView(withId(R.id.doneButton)).perform(click())
+        onView(withId(R.id.addReviewTitle)).check(matches(withText(title)))
+    }
+
+    @Test
+    fun nullTitleNoReset() {
+        onView(withId(R.id.reviewNavigationView)).perform(CustomViewActions.navigateTo(R.id.addRestaurantReviewFragment))
+        val comment = "Good"
+        onView(withId(R.id.reviewRatingBar)).perform(performSetRating(ReviewRating.GOOD))
+        onView(withId(R.id.addReviewComment)).perform(typeText(comment))
+        closeSoftKeyboard()
+        onView(withId(R.id.doneButton)).perform(click())
+        onView(withId(R.id.addReviewComment)).check(matches(withText(comment)))
+    }
+
+
+    @Test
+    fun nonNullArgumentsResetsAddReview() {
+        onView(withId(R.id.reviewNavigationView)).perform(CustomViewActions.navigateTo(R.id.addRestaurantReviewFragment))
+        val comment = "Good"
+        val title = "Good title"
+        onView(withId(R.id.reviewRatingBar)).perform(
+            performSetRating(
+                ReviewRating.GOOD
+            )
+        )
+        onView(withId(R.id.addReviewComment)).perform(typeText(comment))
+        closeSoftKeyboard()
+        onView(withId(R.id.addReviewTitle)).perform(typeText(title))
+        closeSoftKeyboard()
+        onView(withId(R.id.doneButton)).perform(click())
+        onView(withId(R.id.addReviewComment)).check(matches(withText("")))
+        onView(withId(R.id.addReviewTitle)).check(matches(withText("")))
+    }
+
+    @Test
+    fun userNotConnectedNoReset() {
+        FakeConnectedUser.loggedIn = false
+        onView(withId(R.id.reviewNavigationView)).perform(CustomViewActions.navigateTo(R.id.addRestaurantReviewFragment))
+        val comment = "Good"
+        val title = "Good title"
+        onView(withId(R.id.reviewRatingBar)).perform(
+            performSetRating(
+                ReviewRating.GOOD
+            )
+        )
+        onView(withId(R.id.addReviewComment)).perform(typeText(comment))
+        closeSoftKeyboard()
+        onView(withId(R.id.addReviewTitle)).perform(typeText(title))
+        closeSoftKeyboard()
+        onView(withId(R.id.doneButton)).perform(click())
+        onView(withId(R.id.addReviewComment)).check(matches(withText(comment)))
+        onView(withId(R.id.addReviewTitle)).check(matches(withText(title)))
+    }
+
+    companion object {
```

Comment:
I  think you could infer a `CustomViewAction` method from that.

Reply:
This is not my code, should I move it anyway to CustomViewAction ?

Reply:
Ah ok, yes this would remove duplication

Reply:
Done.